### DISCUSSION TITLE
Building the libde265.dll used by YUView from Source to avoid an assertion?

### DISCUSSION THREAD

**Discussion by RafaelLaya** [IENT/YUView#496] [General]
I would like to build the libde265.dll that is bundled with YUView for Windows, from source. There is a runtime assertion that makes it really annoying for me to open some files. 

Specifically it's this assert from this file: https://github.com/strukturag/libde265/blob/master/libde265/image.cc

```
  if (chroma_format != de265_chroma_mono && sps) {
    assert(sps->SubWidthC  == SubWidthC);
    assert(sps->SubHeightC == SubHeightC);
  }
```
I know it's an important assertion to check if the subsampling values for the chroma are equal from what it's derived from the format field, but I know it's still able to understand my files regardless. It would help me a lot in the short-term if I didn't have to keep hitting "ignore" into the assertion. And by knowing how to build from source I can add some prints to debug further and see if there's a bug on libde265 or there's a bug on the encoder I'm using, and eventually upstream a patch where needed. 

However, I couldn't find instructions on how to build the DLL on Windows. Has anyone done this?

Thanks!

**Comment by ChristianFeldmann** [comment]
Building libde265 on windows is straightforward using cmake. In the libde265 repo you can see how to build it in the appveyor CI file: https://github.com/strukturag/libde265/blob/master/appveyor.yml
